As far as the research I have done on the internet , Hausdorff distance is the suitable metric for this question.
Hausdorff distance is a way to measure how different two paths are from each other. Imagine you have two sets of points representing paths, it finds the biggest gap between them. It works by checking each point in one path and seeing how far it is from the nearest point in the other path. This makes it different from Chamfer distance.

Hausdorff distance is super useful in computer vision tasks like matching shapes in images. If the distance between two shapes is small, it means they're pretty similar.

Advantages of using Hausdorff distance:
- It gives a big-picture view of how similar two paths are, unlike the straight-line distance (Euclidean distance) which just looks at point-to-point distances.
- It handles paths that twist and turn and overlap each other.
- It looks at both directions, from path A to B and from B to A.

But, there are some downsides too. Hausdorff distance can be thrown off by outliers or small changes in the paths. Even tiny errors can make a big difference in the distance it calculates.

Other distance metrics you might come across include:
- Euclidean Distance: This is just the straight-line distance between points.
- Manhattan Distance: It's like if you're navigating a city grid, adding up the blocks you travel along each street.
- Chebyshev Distance: This measures the biggest difference along any coordinate axis.
- Edit Distance: It counts how many changes you need to make to turn one path into the other.
- Fréchet Distance: This one's a bit quirky—it's like the minimum leash length you'd need to walk two paths at the same time.

And hey, fun fact: a bunch of these metrics are buddies with dynamic programming too!

https://medium.com/@sim30217/hausdorff-distance-a2580a8dcbb2

For the second part of the question , the first thing that came into my mind was dot product. Higher the dot product of two high dimensional vectors , more similar they are. But after searching in google I came to know that there is something called cosine similarity , which is an extended version of dot product (i.e. dotproduct divided by the product of magnitude of it's vectors).
cosine similary is better because , it gives normalised values whereas dot product can give large values and it becomes harder for us to notice the similarity. Hence cosine similarity is the metric used in identifying similar embeddings
Below are few corpus of words and their embeddings and how well they are related to each other can be seen from the values obtained after applying the metrics.

https://medium.com/@arjunprakash027/understanding-cosine-similarity-a-key-concept-in-data-science-72a0fcc57599
https://www.youtube.com/watch?v=wjZofJX0v4M

In [14]:
# below code is GPT generated and  I have modified it.Only the syntax is from ChatGPT , idea to include animals and comapare between them is mine
import gensim.downloader as api
import numpy as np
from sklearn.manifold import TSNE
import plotly.graph_objects as go

# Function to calculate cosine similarity
def cosine_similarity(v1, v2):
    dot_product = np.dot(v1, v2)
    norm_v1 = np.linalg.norm(v1)
    norm_v2 = np.linalg.norm(v2) #norm is the absolute value of the vector or the magnitude of the vector
    return dot_product / (norm_v1 * norm_v2)

#corpus of words
words = ['happiness', 'success', 'sadness', 'failure','man','woman','king','queen' , 'dog', 'cat', 'rabbit']

# Convert these words into embeddings using a pre-trained word embedding model
# Load pre-trained word vectors
model = api.load("glove-wiki-gigaword-100")
embeddings = np.array([model[word] for word in words])
tsne = TSNE(n_components=3, random_state=42 , perplexity = 3)
embeddings_3d = tsne.fit_transform(embeddings)
fig = go.Figure()

for i, word in enumerate(words):
    fig.add_trace(go.Scatter3d(
        x=[embeddings_3d[i, 0]],
        y=[embeddings_3d[i, 1]],
        z=[embeddings_3d[i, 2]],
        text=[word],
        mode='markers+text',
        marker=dict(size=8),
        textposition='top center'
    ))

fig.update_layout(
    title='Word Embeddings in 3D space',
    scene=dict(
        xaxis_title='Dimension 1',
        yaxis_title='Dimension 2',
        zaxis_title='Dimension 3'
    )
)

fig.show()


In [15]:
similarity = [[0]*len(words) for _ in range(len(words))]
for i in range(len(words)):
    for j in range(len(words)):
        similarity[i][j] = cosine_similarity(embeddings[i], embeddings[j])
        print(f"Cosine similarity between '{words[i]}' and '{words[j]}': {similarity[i][j]:.4f}")


Cosine similarity between 'happiness' and 'happiness': 1.0000
Cosine similarity between 'happiness' and 'success': 0.5104
Cosine similarity between 'happiness' and 'sadness': 0.6159
Cosine similarity between 'happiness' and 'failure': 0.2918
Cosine similarity between 'happiness' and 'man': 0.3360
Cosine similarity between 'happiness' and 'woman': 0.3397
Cosine similarity between 'happiness' and 'king': 0.2362
Cosine similarity between 'happiness' and 'queen': 0.2668
Cosine similarity between 'happiness' and 'dog': 0.2536
Cosine similarity between 'happiness' and 'cat': 0.2405
Cosine similarity between 'happiness' and 'rabbit': 0.1813
Cosine similarity between 'success' and 'happiness': 0.5104
Cosine similarity between 'success' and 'success': 1.0000
Cosine similarity between 'success' and 'sadness': 0.2767
Cosine similarity between 'success' and 'failure': 0.5935
Cosine similarity between 'success' and 'man': 0.4290
Cosine similarity between 'success' and 'woman': 0.3300
Cosine similar

In [16]:
similarity = np.array(similarity)
print(similarity)

[[1.         0.51042324 0.61585224 0.29178962 0.33595636 0.33970976
  0.23624037 0.26679105 0.25360554 0.24054927 0.18125434]
 [0.51042324 1.         0.27665427 0.5934819  0.42904678 0.32999048
  0.3724138  0.36779028 0.2847064  0.25161365 0.14530987]
 [0.61585224 0.27665427 0.99999994 0.36238217 0.24028894 0.21511696
  0.22652636 0.19501376 0.09017212 0.06917604 0.05332812]
 [0.29178962 0.5934819  0.36238217 0.99999994 0.35255626 0.3041968
  0.34956864 0.23595631 0.22314484 0.17546944 0.06423179]
 [0.33595636 0.42904678 0.24028894 0.35255626 1.0000001  0.83234936
  0.5118681  0.47403234 0.5643127  0.52618414 0.3520111 ]
 [0.33970976 0.32999048 0.21511696 0.3041968  0.83234936 1.
  0.36574456 0.5095154  0.49806648 0.47835097 0.3059358 ]
 [0.23624037 0.3724138  0.22652636 0.34956864 0.5118681  0.36574456
  1.0000001  0.7507691  0.29510066 0.32823765 0.26456177]
 [0.26679105 0.36779028 0.19501376 0.23595631 0.47403234 0.5095154
  0.7507691  1.         0.3395347  0.3810363  0.250774  ]
 [

Point to note: cosine similarity between king and queen is 0.7508 and between man and woman is 0.8323 which are closer enough .
Between dog and cat is 0.8798 (famous pets)  and between dog and rabbit is 0.6469 and between cat and rabbit is 0.7424   which are quite high indicating that all animal embeddings are in one direction. Happiness and success lie on one side and failure and sadness lie on the opposite side 